addon/test-waiter.ts

```diff
@@ -14,12 +13,14 @@ function getNextToken(): number {
  * @public
  * @class TestWaiter<T>
  */
-export default class TestWaiter<T = Token> implements ITestWaiter<T> {
+export default class TestWaiter<T extends object | number = Token> implements ITestWaiter<T> {
   public name: WaiterName;
   private nextToken: () => T;
   private isRegistered = false;
 
   items = new Map<T, ITestWaiterDebugInfo>();
+  completedOperationsForTokens = new WeakMap<Token, boolean>();
+  completedOperationsForNumbers = new Map<number, boolean>();
```

Comment:
Technically, this is for anything that is not an object or function right?

Reply:
It _is_, but I believe you and I discussed that we want to phase out allowing users to supply their own tokens, right? Based on that, this is intended to be a temporary solution to still maintain backwards compatibility with `number` types for tokens, while driving people over to using the `Token` type, which is specifically what we want them to use. Is this correct?

Reply:
Discussed offline. We want to allow for primitives as token types, and as such I've updated the type to be `Primative`.